web/react/components/filtered_user_list.jsx

```diff
@@ -37,10 +44,47 @@ class FilteredUserList extends React.Component {
         }
     }
 
-    handleFilterChange(e) {
-        this.setState({
-            filter: e.target.value
-        });
+    searchProfiles() {
```

Comment:
I think we should move this logic out of here and just add an onFilterChange event to the FilteredUserList. It shouldn't need to care at all about where the data came from and whether or not there's more of it. Just updating the filter state and rerendering should be fine while its parent goes off and does whatever it needs to update the source data.